__init__.py

```diff
@@ -0,0 +1,326 @@
+        """
+        self._handle_future_time(request)
+
+    @intent_handler("what-time-will-it-be.intent")
+    def handle_future_time_padatious(self, request: Message):
+        """Respond to a less common request for the future time.
+
+        Example: "when is it 8 hours from now?"
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        self._handle_future_time(request)
+
+    @intent_handler(AdaptIntent("mark-one-idle").require("display").require("time"))
+    def handle_show_time(self, _):
+        """Respond to a request top show the time on a Mark I when idle.
+
+        Example: "What time will it be in 8 hours?"
+        """
+        self.settings["show_time"] = True
+        self.schedule_repeating_event(self.display_idle, None, 10, name="IdleTime")
+
+    def _handle_future_time(self, request: Message):
+        """Respond to a request for the future time.
+
+        Determine the future time, then speak the result to the user and display
+        it (if applicable).
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        response = Response(self.config_core, self.location_regex_path)
+        try:
+            response.build_future_time_response(request.data["utterance"])
+        except LocationNotFoundError:
+            self._handle_location_not_found(response)
+        else:
+            if response.date_time is None:
+                self._handle_current_time(request)
+            else:
+                self._respond(response)
+
+    def _handle_current_time(self, request: Message):
+        """Respond to a request for the current time.
+
+        Determine the current time, then speak the result to the user and display
+        it (if applicable).
+
+        Args:
+            request: Data from the intent parser regarding the user's voice request.
+        """
+        self.log.info("request data: " + str(request.data.keys()))
+        response = Response(self.config_core, self.location_regex_path)
+        try:
+            response.build_current_time_response(request.data["utterance"])
+        except LocationNotFoundError:
+            self._handle_location_not_found(response)
+        else:
+            self._respond(response)
+
+    def _handle_location_not_found(self, response):
+        """User requested time in a city not recognized by a Geolocation API call."""
+        dialog_data = dict(location=response.requested_location)
+        self.speak_dialog("location-not-found", dialog_data)
+
+    def _respond(self, response: Response):
+        """Speak and display the response to the user's request.
+
+        Args
+            response: object used to formulate the response
+        """
+        self._display_time(response)
+        self.speak_dialog(response.dialog_name, response.dialog_data, wait=True)
```

Comment:
This has nothing to do with this PR but I wanted to get it on record somewhere. I am seeing this pattern pop up a lot now where we hard sleep some constant time. This belongs above this skill, maybe in an exit routine and the value should be a system wide config value. I suspect it will need to be in the skill base class as the display handle probably can't be communicated properly in a message.

Reply:
In this case I added the sleep because I clear the display explicitly in the skill in the `_clear_display()` method.  This was done because I wanted different behavior than the default.  I can add a TODO to write some library code on handling clearing the display (Mark I or Mark II)